File: in_ruby/day_01_part_2.rb
require 'set'

input = File.read("../inputs/day01.txt").lines.map(&:to_i).to_set

solution = nil
input.each do |a|
  input.each do |b|
    if input.include?(2020 - a - b)
      solution = a * b * (2020 - a - b)
      break
    end
  end
end

#--
puts "Day 01 - Part 2: #{solution}"

File: in_clojure/day_01_part_2.clj
#!/usr/bin/env bb

(require '[clojure.string :as str])

(def input
  (->> "../inputs/day01.txt"
       (slurp)
       (str/split-lines)
       (map read-string)
       set))

(def solution
  (some identity
        (for [a input b (disj input a)]
          (when-let [c ((disj input a b)
                        (- 2020 a b))]
            (* a b c)))))

(time
 (println "Day 01 - Part 2:" solution))
